internal/resolver/delegatingresolver/delegatingresolver_ext_test.go

```diff
@@ -0,0 +1,447 @@
+	hpfe := func(req *http.Request) (*url.URL, error) { return nil, nil }
+	originalhpfe := internal.HTTPSProxyFromEnvironmentForTesting
+	internal.HTTPSProxyFromEnvironmentForTesting = hpfe
+	defer func() {
+		internal.HTTPSProxyFromEnvironmentForTesting = originalhpfe
+	}()
+
+	mr := manual.NewBuilderWithScheme("test") // Set up a manual resolver to control the address resolution.
+	target := mr.Scheme() + ":///" + targetTestAddr
+
+	// Create a delegating resolver with no proxy configuration
+	tcc, stateCh, _ := createTestResolverClientConn(t)
+	if _, err := delegatingresolver.New(resolver.Target{URL: *testutils.MustParseURL(target)}, tcc, resolver.BuildOptions{}, mr, false); err != nil {
+		t.Fatalf("Failed to create delegating resolver: %v", err)
+	}
+
+	// Update the manual resolver with a test address.
+	mr.UpdateState(resolver.State{
+		Addresses:     []resolver.Address{{Addr: resolvedTargetTestAddr1}},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	})
+
+	// Verify that the delegating resolver outputs the same addresses, as returned
+	// by the target resolver.
+	wantState := resolver.State{
+		Addresses:     []resolver.Address{{Addr: resolvedTargetTestAddr1}},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	}
+
+	var gotState resolver.State
+	select {
+	case gotState = <-stateCh:
+	case <-time.After(defaultTestTimeout):
+		t.Fatal("Timeout when waiting for a state update from the delegating resolver")
+	}
+
+	if !cmp.Equal(gotState, wantState) {
+		t.Fatalf("Unexpected state from delegating resolver: %s, want %s", pretty.ToJSON(gotState), pretty.ToJSON(wantState))
+	}
+}
+
+// setupDNS registers a new manual resolver for the DNS scheme, effectively
+// overwriting the previously registered DNS resolver. This allows the test to
+// mock the DNS resolution for the proxy resolver. It also registers the
+// original DNS resolver after the test is done.
+func setupDNS(t *testing.T) *manual.Resolver {
+	mr := manual.NewBuilderWithScheme("dns")
+
+	dnsResolverBuilder := resolver.Get("dns")
+	resolver.Register(mr)
+
+	t.Cleanup(func() { resolver.Register(dnsResolverBuilder) })
+	return mr
+}
+
+// proxyAddressWithTargetAttribute creates a resolver.Address for the proxy,
+// adding the target address as an attribute.
+func proxyAddressWithTargetAttribute(proxyAddr string, targetAddr string) resolver.Address {
+	addr := resolver.Address{Addr: proxyAddr}
+	addr = proxyattributes.Populate(addr, proxyattributes.Options{
+		ConnectAddr: targetAddr,
+	})
+	return addr
+}
+
+// Tests the scenario where proxy is configured and the target URI contains the
+// "dns" scheme and target resolution is enabled. The test verifies that the
+// addresses returned by the delegating resolver combines the addresses
+// returned by the proxy resolver and the target resolver.
+func (s) TestDelegatingResolverwithDNSAndProxyWithTargetResolution(t *testing.T) {
+	hpfe := func(req *http.Request) (*url.URL, error) {
+		if req.URL.Host == targetTestAddr {
```

Comment:
We should call `t.Errorf` log an error and return to fail the test if we get an unexpected `Host`.